server/src/main/java/org/opensearch/gateway/remote/model/RemoteReadResult.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway.remote.model;
+
+import org.opensearch.core.xcontent.ToXContent;
+
+/**
+ * Container class for entity read from remote store
+ */
+public class RemoteReadResult {
```

Comment:
why do you need this class?

Reply:
We are using this class to hold information after reading the metadata in parallel. Similar to `UploadedIndexMetadata`, `UploadedMetadataAttribtue` we are using while doing parallel writes.